### ISSUE TITLE
Possible performance improvements.

### ISSUE THREAD

**Issue by jdalton** [json5/json5#164]
Just a heads up that I stumbled upon [JSON6](https://github.com/d3x0r/json6) project, inspired by JSON5, but claims to be 2x faster than JSON5. Pinging @d3x0r for details on the perf improvements. My guess is that it deals with char codes instead of string comparisons but I could be off.

**Comment by d3x0r** [comment]
Yes, I'm pretty sure that's where the improvement was.  The initial port of C code was string comparisons and it was about half the speed it is now.
